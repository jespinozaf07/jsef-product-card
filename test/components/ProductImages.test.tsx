import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe de mostrar el componente correctamente con la img personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img={product2.img} className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componente con la img del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
