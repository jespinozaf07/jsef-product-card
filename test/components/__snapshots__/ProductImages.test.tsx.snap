// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la img del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la img personalizada 1`] = `
<img
  className="productImg custom-class"
  src="./coffee-mug.png"
/>
`;
